pkg/ui/src/views/jobs/duration.tsx

```diff
@@ -26,20 +26,28 @@ import {cockroach} from "src/js/protos";
 
 export class Duration extends React.PureComponent<{ job: Job }> {
   render() {
-    const started = TimestampToMoment(this.props.job.started);
-    const finished = TimestampToMoment(this.props.job.finished);
-    const modified = TimestampToMoment(this.props.job.modified);
     if (jobHasOneOfStatuses(this.props.job, JOB_STATUS_PENDING)) {
       return "Waiting for GG TCL";
-    } else if (jobHasOneOfStatuses(this.props.job, JOB_STATUS_RUNNING)) {
+    }
+    // Parse timestamp to default value NULL instead of Date.now.
+    // Conversion dates to Date.now causes traling dates and constant
+    // duration increase even when job is finished.
+    const started = TimestampToMoment(this.props.job.started, null);
+    const modified = TimestampToMoment(this.props.job.modified, null);
+    const finished = TimestampToMoment(this.props.job.finished, null);
+
+    if (!started) {
+      return null;
+    }
+    if (jobHasOneOfStatuses(this.props.job, JOB_STATUS_RUNNING) && !!modified) {
```

Comment:
This component is getting a bit hard to understand with so many if statements, can we simplify?

I think we can remove `jobHasOneOfStatuses`. It's only used here and we always just check 1 status at a time. Maybe a switch statement can help here?